pkg/proxy/proxy.go

```diff
@@ -66,7 +68,9 @@ func SetupDaemonSet(obj *unstructured.Unstructured) error {
 func SetupPod(obj *unstructured.Unstructured) error {
 
 	containers, found, err := unstructured.NestedSlice(obj.Object, "spec", "containers")
-	exit.OnErrorOrNotFound(found, err)
+	if err != nil || !found {
+		return err
```

Comment:
Distinguish between error and not found? If its not found err is nil and this is returned as if it worked ok.

Reply:
Oh indeed, well spotted.